lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1519,40 +1519,32 @@ void ConstantOp::build(OpBuilder &builder, OperationState &result,
   return build(builder, result, type, attr);
 }
 
+static LogicalResult verifySubfieldOp(SubfieldOp op) {
+  if (op.fieldIndex() >
+      op.input().getType().cast<BundleType>().getNumElements())
+    return op.emitOpError("Subfield field index is greater than number of "
+                          "fields in the bundle type");
+  return success();
+}
+
 FIRRTLType SubfieldOp::inferReturnType(ValueRange operands,
                                        ArrayRef<NamedAttribute> attrs,
                                        Optional<Location> loc) {
   auto inType = operands[0].getType();
-  auto fieldname = getAttr<StringAttr>(attrs, "fieldname");
-
-  if (auto bundleType = inType.dyn_cast<BundleType>()) {
-    auto elt = bundleType.getElement(fieldname.getValue());
-    if (!elt) {
-      if (loc)
-        mlir::emitError(*loc, "unknown field '")
-            << fieldname.getValue() << "' in bundle type " << inType;
-      return {};
-    }
-    // FIRRTL puts flips on element fields, not on the underlying
-    // types.  The result type of a subfield should strip a flip
-    // if one exists.
-    return elt->type;
-  }
+  auto fieldIndex =
+      getAttr<IntegerAttr>(attrs, "fieldIndex").getValue().getZExtValue();
 
-  if (loc)
-    mlir::emitError(*loc, "subfield requires bundle operand");
-  return {};
+  // SubfieldOp verifier checks that the field index is valid with number of
+  // subelements. FIRRTL puts flips on element fields, not on the underlying
+  // types.  The result type of a subfield should strip a flip
+  // if one exists.
+  return inType.cast<BundleType>().getElement(fieldIndex).getValue().type;
```

Comment:
This comment about stripping flips is stale. I think you can just strike the last two sentences.